File: 217. Contains Duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        
        //BRUTE FORCE
        /*ArrayList<Integer> arr =new ArrayList<>();
        for(int i:nums)
        {
            if(arr.contains(i))
            {
                return true;
            }
            else
            {
                arr.add(i);
            }
        }

        return false;

*/

        //HASHSET
        HashSet<Integer> set = new HashSet<>();
        for(int i:nums)
        {
            set.add(i);
        }
        if(set.size()==nums.length)
        {
            return false;
        } 
        
        else
        {
            return true;
        }

        
    }
}

File: python_codes/217. Contains Duplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        
        myset = set()
        for n in nums:
            myset.add(n)
        
        if len(myset)<len(nums):
            return True
        else:
            return False


if __name__ == "__main__":
    sol = Solution()
    arr = [1,2,3,4,5]
    print(sol.containsDuplicate(arr))
    arr = [1,2,3,4,5,1]
    print(sol.containsDuplicate(arr))

